### ISSUE TITLE
Remember volume inbetween boots and "default playback device" change

### ISSUE THREAD

**Issue by Jerrk** [Frosthaven/voicemeeter-windows-volume#1]
First of all, i've been looking for this for years so thanks a lot! :smiley: 

I've run into an issue where after rebooting my pc the volume will always be set to 100%, and because i am not used to it yet i often forget and accidentally blast my eardrums and/or wake up the entire house.
i have the same issue when switching the windows "default playback device" on a new boot where it will always be set to 100%.

So it would be great if you could somehow remember the volume level between boots and/or just give an option "set the default volume level on startup to 20%" or something like that

**Comment by Jerrk** [comment]
if Voiceemeeter Windows Volume is set to run on startup it will launch before VM does (at least on my system). 
This makes it so the volume is changed before VM is launched so it does not work.

I can go around this by putting a 30 second delay in the task scheduler when VWV runs. though not sure if there is a better way.

With that done it works 😄 .

It didn't change the issue where if i change the windows default playback device to a different VM input it is also at 100%
![](https://i.imgur.com/ocYGsyw.gif)

but hopefully that could be similarly changed 🙂

**Comment by Jerrk** [comment]
It seems to change the volume before the audio engine has been started, and as i said in the edit above, it appears the volume is reset to 100% whenever the engine is started/restarted. not sure if this same issue manifests with banana/normal. but i've closed all the programs that i can think of that would have an effect with no impact.

I'm not sure if there is a way to change the default delay on the audio engine start or if that is something you or i have changed at some point.

![](https://i.imgur.com/dnM5DjY.gif)

**Comment by ahs16** [comment]
> On audio engine restart? Yes. ![](https://camo.githubusercontent.com/d338346b0084ee6618683ab2a5e7e59bf116f18328219e3abd53b0cd5ecb9ca0/68747470733a2f2f692e696d6775722e636f6d2f706e4c6c4f6f482e676966)
> 
> On changing default audio device? No ![](https://camo.githubusercontent.com/c3cb42c51800384472a7451cba1e23faa90dd443bdd03406565b421e5eee303b/68747470733a2f2f692e696d6775722e636f6d2f73424a3175586a2e676966)
> 
> I have an [ROG MAXIMUS XI HERO](https://rog.asus.com/us/motherboards/rog-maximus/rog-maximus-xi-hero-wi-fi-model/helpdesk_download) motherboard using the onboard audio. I'm using the default windows audio drivers since i've always had issues trying to use their specified realtek ones.

Same thing happens to me.

**Comment by ahs16** [comment]
> On audio engine restart? Yes. ![](https://camo.githubusercontent.com/d338346b0084ee6618683ab2a5e7e59bf116f18328219e3abd53b0cd5ecb9ca0/68747470733a2f2f692e696d6775722e636f6d2f706e4c6c4f6f482e676966)
> 
> On changing default audio device? No ![](https://camo.githubusercontent.com/c3cb42c51800384472a7451cba1e23faa90dd443bdd03406565b421e5eee303b/68747470733a2f2f692e696d6775722e636f6d2f73424a3175586a2e676966)
> 
> I have an [ROG MAXIMUS XI HERO](https://rog.asus.com/us/motherboards/rog-maximus/rog-maximus-xi-hero-wi-fi-model/helpdesk_download) motherboard using the onboard audio. I'm using the default windows audio drivers since i've always had issues trying to use their specified realtek ones.

Same thing happens to me, I'm also using the default windows audio drivers since my headset is connected via USB not a 3.5mm jack.

The following is my headset and motherboard.
- [HyperX Cloud Revolver S](https://www.hyperxgaming.com/us/headsets/cloud-revolver-gaming-headset)
- [ASRock B365M-HDV](https://asrock.com/MB/Intel/B365M-HDV/index.asp)

**Comment by ahs16** [comment]
> On audio engine restart? Yes. ![](https://camo.githubusercontent.com/d338346b0084ee6618683ab2a5e7e59bf116f18328219e3abd53b0cd5ecb9ca0/68747470733a2f2f692e696d6775722e636f6d2f706e4c6c4f6f482e676966)
> 
> On changing default audio device? No ![](https://camo.githubusercontent.com/c3cb42c51800384472a7451cba1e23faa90dd443bdd03406565b421e5eee303b/68747470733a2f2f692e696d6775722e636f6d2f73424a3175586a2e676966)
> 
> I have an [ROG MAXIMUS XI HERO](https://rog.asus.com/us/motherboards/rog-maximus/rog-maximus-xi-hero-wi-fi-model/helpdesk_download) motherboard using the onboard audio. I'm using the default windows audio drivers since i've always had issues trying to use their specified realtek ones.

Same thing happens to me, I'm also using the default windows audio driver which is version **10.0.19041.1202** to be specific.
![image](https://user-images.githubusercontent.com/89221636/137644256-22165b84-9561-46c4-8f89-12dcd0a45603.png)


The following is my headset and motherboard.
- [HyperX Cloud Revolver S](https://www.hyperxgaming.com/us/headsets/cloud-revolver-gaming-headset)
- [ASRock B365M-HDV](https://asrock.com/MB/Intel/B365M-HDV/index.asp)

**Comment by Frosthaven** [comment]
Alright friends, I come with news. After a lot of exhausting research I've reached some conclusions and have also uploaded an update over on the [releases](https://github.com/Frosthaven/voicemeeter-windows-volume/releases/) page.

As it turns out, there is quite a rabbit hole of reports where users are getting 100% volume spikes under Windows. This applet never explicitly sets the volume to 100, so it seems to be an issue between Windows and some audio device drivers.

The goal of this applet is to workaround software limitations, and to those ends I've uploaded a new version for you to test. Try out the new "Prevent 100% Volume Spikes" toggle and let me know if this is a viable solution!

As always, thanks for your contributions!!!
